src/mlpack/tests/neat_test_tools.hpp

```diff
@@ -0,0 +1,296 @@
+          !input(1, i))};
+      error += std::pow(answer[0] - output[0], 2);
+    }
+    return 4 - error;
+  }
+
+  template <typename Archive>
+  void serialize(Archive& /* ar */, const unsigned int /* version */)
+  { /* Nothing to do here */}
+};
+
+/**
+ * A Task class that wraps a continuous RL environment.
+ */
+template<class EnvironmentType>
+class ContinuousRLTask
+{
+ public:
+  ContinuousRLTask(const EnvironmentType& env) : environment(env)
+  { /* Nothing to do here */ }
+
+  double Evaluate(Genome<HardSigmoidFunction>& genome)
+  {
+
+    double total = 0;
+    for (size_t i = 0; i < 5; i++)
+    {
+      // Set the initial state.
+      typename EnvironmentType::State state = environment.InitialSample();
+      arma::vec output = genome.Evaluate(state.Data());
+      double fitness = 0;
+      size_t k = 0;
+      while (!environment.IsTerminal(state))
+      {
+        // std::cout << "Timestep: " << k << std::endl;
+        k++;
+        // Choose the action to perform.
+        typename EnvironmentType::Action action;
+        action.action[0] = output[0];
+
+        // Use the current action to get the next state.
+        // std::cout << "Getting next state" << std::endl;
+        fitness += environment.Sample(state, action, state);
+
+        // Update the state of the genome for the next step.
+        // std::cout << "Finding output" << std::endl;
+        output = genome.Evaluate(state.Data());
+
+        if (k >= 10000)
+          break;
+      }
+      fitness += 500;
+      total += fitness;
+    }
+
+    std::cout << "Fitness: " << total / 5 << std::endl;
+    return total / 5;
+  }
+
+ private:
+  EnvironmentType environment;
+};
+
+/**
+ * A Task class that wraps a discrete RL environment.
+ */
+template<class EnvironmentType>
+class DiscreteRLTask
+{
+ public:
+  DiscreteRLTask(const EnvironmentType& env) : environment(env)
+  { /* Nothing to do here */ }
+
+  double Evaluate(Genome<HardSigmoidFunction>& genome)
+  {
+    // Set the initial state.
+    typename EnvironmentType::State state = environment.InitialSample();
+    arma::vec output = genome.Evaluate(state.Data());
+
+    double fitness = 0;
+    double total = 0;
+    for (size_t i = 0; i < 5; i++)
+    {
+      while (!environment.IsTerminal(state))
+      {
+        // Choose the action to perform.
+        const int size = EnvironmentType::Action::size;
+        output = arma::clamp(output, 0, size - 1);
+        int actionInt = std::round(output[0]);
+        typename EnvironmentType::Action action = static_cast<typename
+            EnvironmentType::Action>(actionInt);
+
+        // Use the current action to get the next state.
+        fitness += environment.Sample(state, action, state);
+
+        // Update the state of the genome for the next step.
+        output = genome.Evaluate(state.Data());
+
+        if (fitness > 10000)
+          break;
+      }
+      total += fitness;
```

Comment:
We never reset the ``fitness`` value, so this is already the total fitness, I think we should reset the ``fitness`` parameter before each run, so that ``if (fitness > 10000)`` does work as well.

Reply:
Ahh, nevermind I can see you do reset the value.